modules/dirwatcher.js

```diff
@@ -0,0 +1,94 @@
+const EventEmitter = require('events');
+const fs = require('fs');
+const crypto = require('crypto');
+
+class DirWatcher extends EventEmitter {
+    constructor(path, delay) {
+        super();
+
+        this.path = path;
+        this.delay = delay;
+        this.countFiles = 0;
+        this.files = [];
+
+        this.interval = setInterval(() => {
+            console.log('WATCH: ' + new Date().toISOString().replace(/T/, ' ').replace(/\..+/, ''));
+            this.watch();
+        }, this.delay)
+    }
+
+    difference(arr1, arr2) {
+        var result = arr1.filter(e => !arr2.includes(e));
+        return result;
+    }
+
+    createHash(filename) {
+        const hash = crypto.createHash('sha256');
+
+        const fileContent = fs.readFileSync(filename);
+        hash.update(fileContent);
+        const result = hash.digest('hex');
+        return result;
+    }
+
+    watch() {
+        var dirwatcher = this;
```

Comment:
I wonder why do you need to save the context? You can use arrow function if you need this in the callback